xclim/indices/fire/_ffdi.py

```diff
@@ -0,0 +1,402 @@
+    smd="[precipitation]",
+)
+def griffiths_drought_factor(
+    pr: xr.DataArray,
+    smd: xr.DataArray,
+    limiting_func: str = "xlim",
+) -> xr.DataArray:
+    """Griffiths drought factor based on the soil moisture deficit.
+
+    The drought factor is a numeric indicator of the forest fire fuel availability in the
+    deep litter bed. It is often used in the calculation of the McArthur Forest Fire Danger
+    Index. The method implemented here follows :cite:t:`ffdi-finkele_2006`.
+
+    Parameters
+    ----------
+    pr : xr.DataArray
+      Total rainfall over previous 24 hours [mm/day].
+    smd : xarray DataArray
+      Daily soil moisture deficit (often KBDI) [mm/day].
+    limiting_func : {"xlim", "discrete"}
+      How to limit the values of the drought factor.
+      If "xlim" (default), use equation (14) in :cite:t:`ffdi-finkele_2006`.
+      If "discrete", use equation Eq (13) in :cite:t:`ffdi-finkele_2006`, but with the lower
+      limit of each category bound adjusted to match the upper limit of the previous bound.
+
+    Returns
+    -------
+    df : xr.DataArray
+      The limited Griffiths drought factor.
+
+    Notes
+    -----
+    Calculation of the Griffiths drought factor depends on the rainfall over the previous 20 days.
+    Thus, the first non-NaN time point in the drought factor returned by this function
+    corresponds to the 20th day of the input data.
+
+    References
+    ----------
+    :cite:cts:`ffdi-griffiths_1999,ffdi-finkele_2006,ffdi-holgate_2017`
+    """
+
+    def _griffiths_drought_factor_pass(pr, smd, lim):
+        """Pass inputs on to guvectorized function `_griffiths_drought_factor`. DO NOT CALL DIRECTLY, use `griffiths_drought_factor` instead."""
+        # This function is actually only required as xr.apply_ufunc will not receive
+        # a guvectorized function which has the output(s) in its function signature
+        return _griffiths_drought_factor(pr, smd, lim)
+
+    pr = convert_units_to(pr, "mm/day")
+    smd = convert_units_to(smd, "mm/day")
+
+    if limiting_func == "xlim":
+        lim = 0
+    elif limiting_func == "discrete":
+        lim = 1
+    else:
+        raise ValueError(f"{limiting_func} is not a valid input for `limiting_func`")
+
+    df = xr.apply_ufunc(
+        _griffiths_drought_factor_pass,
+        pr,
+        smd,
+        kwargs=dict(lim=lim),
+        input_core_dims=[["time"], ["time"]],
+        output_core_dims=[["time"]],
+        dask="parallelized",
+        output_dtypes=[pr.dtype],
+    )
+    df.attrs["units"] = ""
+
+    # First non-zero entry is at the 19th time point since df is calculated
+    # from a 20-day rolling window. Make prior points NaNs.
+    return df.where(df.time >= df.time.isel(time=19))
+
+
+@declare_units(
+    D="[]",
+    T="[temperature]",
+    H="[]",
+    V="[speed]",
+)
+def mcarthur_forest_fire_danger_index(
+    D: xr.DataArray,
+    T: xr.DataArray,
+    H: xr.DataArray,
+    V: xr.DataArray,
+):  # noqa: D403
```

Comment:
Would it be possible to use more common variable names? I understand the definitions used here are a bit different, but they seem close enough?
Like: D->  drought_factor, or df
T -> tasmax
H -> hurs or hursmax
V -> sfcWind
?